### PR TITLE
WIP: Added test to see if script can connect to ldap

### PR DESCRIPTION
**Pull request by gruunday**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -6,6 +6,8 @@ services:
       - internal
     depends_on:
       - ldap
+    links:
+      - useradm:ldap
```
Comment (butlerx) [inline docker-compose.yml]: you don't need this with depends_on

**Review by butlerx — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -12,9 +12,10 @@ jobs:
           SLAPD_PASSWORD: 'redbrick'
           SLAPD_DOMAIN: 'ldap.internal'
     steps:
-      - run: apk add --no-cache openldap-dev cyrus-sasl-dev git build-base
+      - run: apk add --no-cache openldap-dev openldap-clients cyrus-sasl-dev git build-base
       - checkout
       - run: pip install -r useradm/requirements.txt
+      - run: ldapsearch -x -h ldap.internal -b "ou=accounts,o=redbrick"
```
Comment (butlerx) [inline .circleci/config.yml]: where are you defining the network `.internal`?

File: docker-compose.yml [hunk 3]
```diff
@@ -16,6 +18,8 @@ services:
       SLAPD_DOMAIN: 'ldap.redbrick'
     volumes:
       - ./ldif:/etc/ldap.dist/prepopulate
+    links:
+      - useradm:ldap
```
Comment (butlerx) [inline docker-compose.yml]: same as above

File: .circleci/config.yml [hunk 4]
```diff
@@ -12,9 +12,10 @@ jobs:
           SLAPD_PASSWORD: 'redbrick'
           SLAPD_DOMAIN: 'ldap.internal'
     steps:
-      - run: apk add --no-cache openldap-dev cyrus-sasl-dev git build-base
+      - run: apk add --no-cache openldap-dev openldap-clients cyrus-sasl-dev git build-base
```
Comment (butlerx) [inline .circleci/config.yml:15]: should this be added to the dockerfile aswell?

File: docker-compose.yml [hunk 5]
```diff
@@ -6,16 +6,22 @@ services:
...
       - ldap
+    links:
+      - useradm:ldap
   ldap:
     image: dinkel/openldap
     container_name: ldap
     networks:
       - internal
+    ports:
+      - "389:389"
```
Comment (butlerx) [inline docker-compose.yml]: dont expose this on the host

File: docker-compose.yml (hunk 1 above)
Comment (gruunday) [reply docker-compose.yml]: Are you sure? I have seen configs with both?

File: .circleci/config.yml (hunk 2 above)
Comment (gruunday) [reply .circleci/config.yml]: https://github.com/redbrick/useradm/blob/master/.circleci/config.yml#L13

File: .circleci/config.yml (hunk 4 above)
Comment (gruunday) [reply .circleci/config.yml:15]: Does it need to be?

File: docker-compose.yml (hunk 1 above)
Comment (butlerx) [reply docker-compose.yml]: links is from version 1 and was replaced with depend_on in v2 and is better as it handles both parts of the linl and makes sure the containers are started in the right order

File: docker-compose.yml (hunk 5 above)
Comment (gruunday) [reply docker-compose.yml]: I am getting errors. I am trying to fix those errors. Testing all options

File: .circleci/config.yml (hunk 2 above)
Comment (butlerx) [reply .circleci/config.yml]: thast just setting the domain name of the server in ldap for the schema not actually creating a netowork

File: .circleci/config.yml (hunk 4 above)
Comment (butlerx) [reply .circleci/config.yml:15]: :man_shrugging:

File: .circleci/config.yml (hunk 2 above)
Comment (gruunday) [reply .circleci/config.yml]: What would it be then? Local host

Comment (butlerx) [reply .circleci/config.yml]: https://circleci.com/docs/2.0/executor-types/#using-multiple-docker-images

Comment (gruunday) [reply .circleci/config.yml]: Right, so localhost does not work. Any other ideas as to why they can't connect?

Comment (butlerx) [reply .circleci/config.yml]: try `telnet localhost 389`

File: .circleci/config.yml [hunk 6]
```diff
@@ -12,9 +12,11 @@ jobs:
...
       - checkout
       - run: pip install -r useradm/requirements.txt
+      - run: ldapsearch -x -L -h localhost -D admin -w redbrick -b "dc=ldap,dc=internal"
+      - run: telnet localhost:389
```
Comment (butlerx) [inline .circleci/config.yml]: telnet first

Comment (gruunday) [reply .circleci/config.yml]: Yes but getting new error now :P

**Pull request closed without merging**